script/hassfest/dependencies.py

```diff
@@ -297,17 +313,21 @@ def _check_circular_deps(
     checking.remove(integration.domain)
 
 
-def validate(integrations: dict[str, Integration], config: Config) -> None:
-    """Handle dependencies for integrations."""
-    # check for non-existing dependencies
+def _validate_circular_dependencies(integrations: dict[str, Integration]) -> None:
     for integration in integrations.values():
-        validate_dependencies(
-            integrations,
-            integration,
-            check_dependencies=not config.specific_integrations,
+        if integration.domain in IGNORE_VIOLATIONS:
+            continue
+
+        _check_circular_deps(
+            integrations, integration.domain, integration, set(), deque()
         )
 
-        if config.specific_integrations:
+
+def _validate_dependencies_exist(
```

Comment:
The function checks both that referenced dependencies exist and that they're not duplicated in dependencies and after_dependencies. Please update the function name and the comment accordingly